app/jobs/storage/materialize_job.rb

```diff
@@ -0,0 +1,8 @@
+class Storage::MaterializeJob < ApplicationJob
+  queue_as :backend
+  limits_concurrency to: 1, key: ->(owner) { "storage_materialize:#{owner.to_gid}" }
```

Comment:
If you returned `owner` directly as the key it would generate the key for your based on its ID an the job class name. That would work here, right?